activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java

```diff
@@ -249,6 +250,7 @@ public void writePayload(Metadata object, DataOutput dataOut) throws IOException
     protected PageFile pageFile;
     protected Journal journal;
     protected Metadata metadata = new Metadata();
+    protected PersistenceAdapterStatistics persistenceAdapterStatistics = new PersistenceAdapterStatistics();
```

Comment:
I would probably make this final and then I would also add a getter for this object and also expose it under KahaDBPersistenceAdapter. I routinely use embedded brokers for example and being able to access this through KahaDBPersistenceAdapter without JMX would be nice.  Inside KahaDBPersistenceAdapter I would just return the actual PersistenceAdapterStatistics object so someone could access it.

Reply:
Done.